src/Maker/MakeRegistrationForm.php

```diff
@@ -368,7 +369,15 @@ private function getMissingComponentsComposerMessage(): ?string
         $missing = false;
         $composerMessage = 'composer require';
 
-        if (!class_exists(SymfonyCastsVerifyEmailBundle::class)) {
+        // verify-email-bundle 1.2 includes support for translations and a fix for the bad expiration time bug.
+        // we need to check that if the bundle is installed, it is version 1.2 or greater
+        if (class_exists(SymfonyCastsVerifyEmailBundle::class)) {
+            $reflectedComponents = new \ReflectionClass(VerifyEmailSignatureComponents::class);
+
+            if (!$reflectedComponents->hasMethod('getExpirationMessageKey')) {
+                throw new RuntimeCommandException('Please upgrade symfonycasts/verify-email-bundle to version 1.2.0 or greater.');
+            }
```

Comment:
Hmm, I think this would cause the entire commend to explode *right* at the end of it working. I didn't really think about that. Probably we should just return a strongly-worded message here to show to the user (instead of `composer require symfonycasts/..`) or we need to do this check *right* when the command is starting, before we ask any questions (the idea would be that, if they have verifier already installed, we will assume they probably ARE going to generate with email verification, so we'll stop them immediately and tell them to upgrade).

Reply:
Currently the very first question that gets asked is "Do you want to verify emails?..." (paraphrasing) - then internally we:

- check if the bundle is installed. If it is not we: (Original Logic)
![image](https://user-images.githubusercontent.com/40327885/102545669-9388aa00-4084-11eb-8f43-6d4c04796035.png)

- If the bundle is installed but does not have the new trans + expiresAt fix we :
![image](https://user-images.githubusercontent.com/40327885/102545847-d0ed3780-4084-11eb-993d-476df8f414fd.png)

The latter is how we are handling it in `make:reset`. The same type of exception + "halt" is done if `reset-password-bundle` is not installed and you try to run the `make:reset-password` command.

For verify, I think the "exception + halt" strategy would be best if someone runs `make:registration-form` -> `y` to the verify email question while already having the old bundle installed.

Reply:
For clarity in the future - the above screen shot shows `...version 1.2.0...` the actual release is `1.1.1`